### PR TITLE
Patient fix

### PR DESCRIPTION
**Pull request by taylorleh**
- fixes double submit on get-finded form
- fixes error xhr error by adding ng-src to img src attribute on patients view. client would request image source before controller had loaded patient records
- redirects patient to new profile page after successful form completion
- changed fileupload factory to return $http - this allows the use of promises in the controllers

WORKING ON: handling and updating view on mysql errors for get-funded controller

### CODE REVIEW

**Comment by taylorleh** [comment]
@benjaminhoffman

**Comment by taylorleh** [comment]
@benjaminhoffman okay I made the changes. I removed the success and error from factory to the controller. this will allows the controller to provide the appropriate response to new-patient POST succeeding of failing.

**Comment by taylorleh** [comment]
@benjaminhoffman let me know if you want me to go over this

**Comment by benjaminhoffman** [comment]
@taylorleh looks like the only change from was a minor one in factory.js?  the others we already reviewed, yeah?

**Comment by taylorleh** [comment]
@benjaminhoffman I removed succes and error from factory and instead allowing the controller to 'then' and 'catch' the response from the new patient POST

File: client/services/factory.js [hunk 1]
```diff
@@ -2,25 +2,23 @@ var app = angular.module('eir.factory', []);
...
+    uploadFileToUrl:function(fields, uploadUrl) {
+      uploadUrl = uploadUrl || 'classes/patients';
+      var fd = new FormData();
+      
+      // append each field to the FormData object
+      for(var field in fields) {
+        fd.append(field, fields[field]);
+      }
+      
+      return $http.post(uploadUrl, fd, {
```
Comment (taylorleh) [inline client/services/factory.js:15]: @benjaminhoffman here I am just returning the http factory... this returns a promise that is thenable and catchable

File: client/patients/get-funded.js [hunk 2]
```diff
@@ -33,7 +16,16 @@ angular.module('eir.getFunded', [])
 
     // only submit if form is valid
     if(form.$valid) {
-      fileUpload.uploadFileToUrl($scope.patient);
+      fileUpload.uploadFileToUrl($scope.patient)
+        .then(function (res, err){
```
Comment (taylorleh) [inline client/patients/get-funded.js:20]: @benjaminhoffman here I am catching the promise response (if there is no error) and redirecting to their new page. this start at line 20 and ends at line 24. #L20-23

File: client/patients/get-funded.js [hunk 3]
```diff
@@ -33,7 +16,16 @@ angular.module('eir.getFunded', [])
...
+          // redirect patient to their new page
+          $location.path('/donate/' + res.data.insertId);
+        })
+        .catch(function (err, res){
```
Comment (taylorleh) [inline client/patients/get-funded.js:24]: @benjaminhoffman here I am catching any and all errors while trying to submit a new patient. these would be
- duplicate username
- invalid email
- mysql server not online
- etc

I still have to handle these errors and this is what I am doing now. For the most part the one error that will likely come back will be duplicate username, which I then need to display a box next to the username with some text like 'already taken, choose new username'. This starts at line 24

**Pull request merged**